src/app/Http/Controllers/Operations/ShowOperation.php

```diff
@@ -75,15 +75,24 @@ public function show($id)
         // get entry ID from Request (makes sure its the last ID for nested resources)
         $id = $this->crud->getCurrentEntryId() ?? $id;
 
+        $this->data['crud'] = $this->crud;
+        $this->data['title'] = $this->crud->getTitle() ?? trans('backpack::crud.preview').' '.$this->crud->entity_name;
+
         // get the info for that entry (include softDeleted items if the trait is used)
-        if ($this->crud->get('show.softDeletes') && in_array('Illuminate\Database\Eloquent\SoftDeletes', class_uses($this->crud->model))) {
-            $this->data['entry'] = $this->crud->getModel()->withTrashed()->findOrFail($id);
-        } else {
+        if (! $this->crud->get('show.softDeletes') || ! in_array('Illuminate\Database\Eloquent\SoftDeletes', class_uses($this->crud->model))) {
             $this->data['entry'] = $this->crud->getEntryWithLocale($id);
+
+            return view($this->crud->getShowView(), $this->data);
         }
 
-        $this->data['crud'] = $this->crud;
-        $this->data['title'] = $this->crud->getTitle() ?? trans('backpack::crud.preview').' '.$this->crud->entity_name;
+        if ($this->crud->get('show.usePanelQuery')) {
+            $this->data['entry'] = $this->crud->query->withTrashed()->findOrFail($id);
+            $this->data['entry'] = $this->crud->setLocaleOnModel($this->data['entry']);
```

Comment:
Uuuu what does THIS button do?!

![working-cartoon-network-gif](https://github.com/Laravel-Backpack/CRUD/assets/1032474/8171cf36-4dbe-4e96-9574-1b7e8b508412)

And why is this needed here? And only here?

Reply:
Fair enough. Writing a descriptive comment for you 🙏

Reply:
Hmm... something here is fishy to me. Why aren't we using `getEntryWithLocale()` and we are using `setLocaleOnModel()`? Looks to me like we're using a setter, instead of a getter.

Reply:
Would this do the same thing or no?

```suggestion
            $this->data['entry'] = $this->crud->getEntryWithLocale($id);
```